packages/web/stores/transfer-history.ts

```diff
@@ -150,16 +173,64 @@ export class NonIbcBridgeHistoryStore implements TransferStatusReceiver {
       return;
     }
 
-    // update balances if successful
-    if (status === "success") {
-      this.queriesStore
-        .get(this.chainId)
-        .queryBalances.getQueryBech32Address(snapshot.accountAddress)
-        .fetch();
-    }
-
+    // set updates
     snapshot.status = status;
     snapshot.reason = reason;
+
+    switch (status) {
+      case "pending":
+        displayToast(
+          {
+            titleTranslationKey: snapshot.isWithdraw
+              ? "transfer.pendingWithdraw"
+              : "transfer.pendingDeposit",
+          },
+          ToastType.LOADING,
+          { toastId: prefixedKey, autoClose: false }
+        );
+        break;
+      case "success":
+        if (this._resolvedTxStatusKeys.has(prefixedKey)) break;
+        displayToast(
+          {
+            titleTranslationKey: snapshot.isWithdraw
+              ? "transfer.completedWithdraw"
+              : "transfer.completedDeposit",
+            captionTranslationKey: snapshot.amount,
+          },
+          ToastType.SUCCESS
+        );
+        this.onAccountTransferSuccess(snapshot.accountAddress);
+        this._resolvedTxStatusKeys.add(prefixedKey);
+        toast.dismiss(prefixedKey);
+        break;
+      case "failed":
+        if (this._resolvedTxStatusKeys.has(prefixedKey)) break;
+        displayToast(
+          {
+            titleTranslationKey: snapshot.isWithdraw
+              ? "transfer.failedWithdraw"
+              : "transfer.failedDeposit",
+            captionTranslationKey: snapshot.amount,
+          },
+          ToastType.ERROR
+        );
+        this._resolvedTxStatusKeys.add(prefixedKey);
+        toast.dismiss(prefixedKey);
+        break;
+      case "connection-error":
+        if (this._resolvedTxStatusKeys.has(prefixedKey)) break;
+        displayToast(
+          {
+            titleTranslationKey: "transfer.connectionError",
+            captionTranslationKey: snapshot.amount,
+          },
+          ToastType.ERROR
+        );
+        this._resolvedTxStatusKeys.add(prefixedKey);
+        toast.dismiss(prefixedKey);
+        break;
+    }
```

Comment:
Thoughts on [updating](https://fkhadra.github.io/react-toastify/update-toast) the existing toast rather than dismissing and displaying new ones?

Reply:
Yeah could be less jarring